cores/esp32/Esp.cpp

```diff
@@ -199,10 +204,10 @@ bool EspClass::flashEraseSector(uint32_t sector)
 
 bool EspClass::flashWrite(uint32_t offset, uint32_t *data, size_t size)
 {
-    return spi_flash_write(offset, (uint32_t*) data, size) == ESP_OK;
+    return spi_flash_write_encrypted(offset, (uint32_t*) data, size) == ESP_OK;
```

Comment:
That makes sense. I suppose that another things which is missing in Arduino for flash encryption support are:
- some kind of way to select whether the bootloader needs to perform flash encryption
- some place to store the unique per-device flash encryption key on the PC (and make it available to different Arduino installations on the same computer?)
- a way to say "I want to enable flash encryption for this board", which will cause the key to be generated and store to the aforementioned location. This part may be implemented using an IDE plugin.
- some diagnostics when a user tries to upload sketch into an ESP which has flash encryption enabled, without having a valid encryption key for this ESP.

Overall, this may be a lot of work. I'm not 100% sure that Arduino users need flash encryption in the first place (which is not even very useful without enabling Secure Boot).